src/cli/extensions/helpers.js

```diff
@@ -0,0 +1,281 @@
+import chalk from 'chalk';
+import resolve from 'resolve';
+import { isString } from 'lodash';
+import semver from 'semver';
+
+import { registerHooks, getHooks, setHooks } from '../../hooks';
+import { registerActions, getActions, setActions } from '../../hooks/actions';
+import { merge } from '../../configuration';
+import { warningLabel, feedbackMessage } from '../../helpers/style';
+import ExtensionError from './error';
+
+export function runPostInits(initialState) {
+    return initialState.postInits.reduce(
+        (state, postInit) =>
+            manageRocObject(
+                postInit({
+                    config: state.config,
+                    meta: state.meta,
+                    extensions: state.usedExtensions,
+                    actions: state.actions,
+                    hooks: state.hooks
+                }), state
+            ),
+        initialState
+    );
+}
+
+export function getExtensions(type) {
+    return (extensions, directory) => (initialState) => {
+        return extensions.reduce(
+            (state, extensionPath) => {
+                // Get the extension
+                const roc = getExtension(extensionPath, directory, type);
+
+                if (roc) {
+                    try {
+                        const nextState = getCompleteExtensionTree(
+                            roc,
+                            extensionPath,
+                            state
+                        );
+
+                        nextState.projectExtensions.push({
+                            name: roc.name,
+                            version: roc.version
+                        });
+
+                        return nextState;
+                    } catch (err) {
+                        console.log(feedbackMessage(
+                            warningLabel('Warning', 'Roc Extension Failed'),
+                            `Failed to load Roc ${type} ${chalk.bold(extensionPath)}\n\n` +
+                            (state.verbose ? err.stack : err.toString()),
+                            err.getPath && err.getPath()
+                        ));
+                    }
+                    // Rest Actions & Hooks
+                    setActions(state.actions);
```

Comment:
Primarily naming. Since it returns a new function it's not the `get`X that mutates anything strictly. It returns functions that will allow it down the line.... I guess just giving the other functions names would help a bit with clarity.